scripts/import/import_variant_synonyms

```diff
@@ -112,39 +111,364 @@ elsif($source_name eq 'dbSNP'){
+    die "Failed to unzip data :$@\n" unless $@ eq '';
+    $output_file = "rsid.tsv"; 
+  }
+  else {
+    $output_file = $data_file; 
+  }
+
+  return $output_file; 
+} 
+
+=head2 parse_uniprot
+
+parse UniProt file 
+
+=cut 
+
+sub parse_uniprot {
+  my $infile = shift;
+  
+  my %synonym;
+    
+  # Open the input file for reading
+  if($infile =~ /gz$/) {
+    open (IN, "zcat $infile |") or die ("Could not open $infile for reading");
+  }
+  else {
+    open (IN,'<',$infile) or die ("Could not open $infile for reading");
+  }
+  
+  # Read through the file and parse out the desired fields
+  while (<IN>) {
+    chomp;
+    
+    # A regexp to catch the meta information in the header. Just echo this to the stdout for logging purposes
+    if ($_ =~ m/^\s*(Description|Name|Release)\:/) {
+      print STDOUT $_ . "\n";
+    }
+    
+    # Main regexp to extract relevant variation information
+    if ($_ =~ m/^(\S+)\s+\S+\s+(VAR\_\d+)\s+\w\.\S+\s+(Disease|Polymorphism|Unclassified)\s+(\-|rs\d*)\s*(.*)$/) {
+      
+      # Get the data that was caught by the regexp
+      my $gene = $1;
+      my $uniprot_id = $2;
+      my $rs_id = $4;
+      my $phenotype = $5;
+      
+      # If no rsId was given, will attempt to get one by looking up the Uniprot id in the synonym table
+      if ($rs_id ne '-') {
+        push(@{$synonym{$rs_id}},$uniprot_id);
+      }
+    }
+  }
+  close(IN);
+  
+  my %result = ('synonyms' => \%synonym);
+  return \%result;
+}
+
+=head2 get_dbIDs
+
+get variation ids 
+
+=cut 
+
+sub get_dbIDs {
+  my $rs_ids     = shift;
+  my $db_adaptor = shift;
+  
+  my $id_stmt = qq{
+    SELECT
+      DISTINCT
+      v.variation_id,
+      v.name
+    FROM
+      variation v,
+      variation_feature vf
+      
+    WHERE
+      v.name = ? AND
+      v.variation_id=vf.variation_id
+    LIMIT 1
+  };
+  my $syn_stmt = qq{
+    SELECT
+      DISTINCT
+      v.variation_id,
+      v.name
+    FROM
+      variation_feature vf,
+      variation_synonym vs JOIN
+      variation v ON vs.variation_id = v.variation_id
+    WHERE
+      vs.name = ? AND
+      v.variation_id=vf.variation_id
+    LIMIT 1
+  };
+  my $id_sth = $db_adaptor->dbc->prepare($id_stmt);
+  my $syn_sth = $db_adaptor->dbc->prepare($syn_stmt);
+  
+  my %mapping;
+  
+  foreach my $rs_id (@{$rs_ids}) {
+    $id_sth->bind_param(1,$rs_id,SQL_VARCHAR);
+    $id_sth->execute();
+    my ($var_id,$var_name);
+    $id_sth->bind_columns(\$var_id,\$var_name);
+    $id_sth->fetch();
+    
+    # If we couldn't find the rs_id, look in the synonym table
+    if (!defined($var_id)) {
+      $syn_sth->bind_param(1,$rs_id,SQL_VARCHAR);
+      $syn_sth->execute();
+      $syn_sth->bind_columns(\$var_id,\$var_name);
+      $syn_sth->fetch();
+    }
+    # warn "$rs_id - no mapping found in variation db\n" unless $var_id ;  
+
+    $mapping{$rs_id} = [$var_id,$var_name] if $var_id && $var_name;
+  }
+  
+  return \%mapping;
+} 
+
+=head2 add_set_uniprot
+
+add uniprot variation set 
```

Comment:
Suggest changing uniprot to UniProt in description to match other sub descriptions